eventlet/wsgi.py

```diff
@@ -879,12 +888,11 @@ def log_message(self, message):
 try:
     import ssl
     ACCEPT_EXCEPTIONS = (socket.error, ssl.SSLError)
-    ACCEPT_ERRNO = {errno.EPIPE, errno.EBADF, errno.ECONNRESET,
```

Comment:
The man page for accept() on Linux says "EBADF:  sockfd is not an open file descriptor." I.e. it's not a recoverable error.

Reply:
Indeed, that seems logic. I think you are right, this kind of error (bad file descriptor), is not recoverable. I don't see how a server could recover from this kind of error.

By doing my previous tests, it seems that in this case (`EBADF`), the `wsgi` server is terminated properly without raising more attention than that.

In parallel, I think we should also consider that the wsgi server is only one piece of the architecture of our users. Some people may launch several greenthreads dedicated to various task, and the wsgi server `greenthread` could be one of them, so if the server exit abruptly, the global process will be affected and then all the other greenthreads would also exit abruptly, then killing all the other background jobs. I think this scenario is more or less the reason behind the implementation of the logic of `EBADF` management we observe in various hubs.

However, this logic more or less hide wsgi server errors. Without digging too far, developer should observe this kind of failure on the client side, when requests are no longer successful.

From a monitoring perspective, if developers monitors the process of their wsgi server, currently, by functionning this way, they should not be able to observe a failure on this process, as the global process could continue to run due to other greenthreads.

I'd argue that when we face this kind of scenario (EBADF), we should failed early as soon as possible.

IMO, rather than implementing specific logic in various hubs, logic dedicated to manage bad file descriptor, the `wsgi` module should exit with error to catch developer attention about this problem.

The current behaviour of hubs seems more a convenience to allow polymorphic architectures in the same process. A kind of fault tolerance behaviour. I'd rather encourage managing this kind of error by using things like [pacemaker](https://clusterlabs.org/pacemaker/).

Thoughts?